src/main/java/com/amazonaws/secretsmanager/sql/AWSSecretsManagerDriver.java

```diff
@@ -376,18 +389,14 @@ public Connection connect(String url, Properties info) throws SQLException {
                 // Most likely to occur in the event that the data is not JSON. This is more of a user error.
                 throw new RuntimeException(INVALID_SECRET_STRING_JSON);
             }
-        }
 
-        if (info != null && info.getProperty("user") != null) {
-            String credentialsSecretId = info.getProperty("user");
             try {
-                return connectWithSecret(unwrappedUrl, info, credentialsSecretId);
+                return connectWithSecret(unwrappedUrl, info, url);
```

Comment:
And in the case a user is using a secret as the jdbc url, just use that value rather than looking in user.